proxy/http/HttpSM.cc

```diff
@@ -7988,3 +7987,58 @@ HttpSM::find_proto_string(HTTPVersion version) const
   }
   return nullptr;
 }
+
+// YTS Team, yamsat Plugin
+// Function to copy the partial Post data while tunnelling
+void
+PostDataBuffers::copy_partial_post_data()
+{
+  this->postdata_copy_buffer->write(this->ua_buffer_reader);
+  Debug("http_redirect", "[PostDataBuffers::copy_partial_post_data] wrote %" PRId64 " bytes to buffers %" PRId64 "",
+        this->ua_buffer_reader->read_avail(), this->postdata_copy_buffer_start->read_avail());
+  this->ua_buffer_reader->consume(this->ua_buffer_reader->read_avail());
+}
+
+// YTS Team, yamsat Plugin
+// Allocating the post data buffers
+void
+PostDataBuffers::init(IOBufferReader *ua_reader)
+{
+  // the this->ua_buffer_reader may be realeased in tunnel::deallocate_buffers.
+  // we always set a new reader instead of the old value;
+  // ink_assert(this->ua_buffer_reader == nullptr);
```

Comment:
Probably we should remove this comment and guarantee that `this->ua_buffer_reader != nullptr` before we return.

The following assume it is not null:
* `PostDataBuffers::copy_partial_post_data`
* `HttpSM::postbuf_read_avail`

Reply:
Sounds reasonable !